app/refresh-ingestion.sh

```diff
@@ -85,11 +85,71 @@ scrape_and_ingest() {
     fi
 
     create_md_zip "$DATASET_ID"
+}
+
+create_md_zip(){
+    local DATASET_ID="$1"
+    [ -d "${DATASET_ID}_md" ] || exit 29
+
+    # Collect stats before zipping
+    local MARKDOWN_COUNT=$(find "${DATASET_ID}_md" -type f -iname '*.md' | wc -l)
+    local INGEST_STATS=$(grep -E "Running with args|DONE splitting|Finished ingesting" "logs/${DATASET_ID}-2ingest.log")
+    local SCRAPE_STATS=$(grep -E 'log_count|item_scraped_count|request_depth|downloader/|httpcache/' "logs/${DATASET_ID}-1scrape.log")
+    local HTML_COUNT=0
+    if [ -d "src/ingestion/imagine_la/scrape/pages" ]; then
+        HTML_COUNT=$(ls src/ingestion/imagine_la/scrape/pages | wc -l)
+    fi
+
+    # Parse ingest stats
+    local PAGES_COUNT=$(echo "$INGEST_STATS" | grep "DONE splitting" | sed -E 's/.*splitting all ([0-9]+) webpages.*/\1/')
+    local CHUNKS_COUNT=$(echo "$INGEST_STATS" | grep "DONE splitting" | sed -E 's/.*total of ([0-9]+) chunks.*/\1/')
+    local INGEST_STATUS=$(echo "$INGEST_STATS" | grep -q "Finished ingesting" && echo "completed" || echo "failed")
+    
+    # Parse scrape stats into key-value pairs
+    local SCRAPE_PARSED=$(echo "$SCRAPE_STATS" | sed 's/{//g; s/}//g; s/'"'"'//g' | tr -d '\n' | sed 's/: /:/g' | tr ',' '\n' | while read -r line; do
+        key=$(echo "$line" | cut -d':' -f1 | tr -d ' ')
+        value=$(echo "$line" | cut -d':' -f2 | tr -d ' ')
+        if [ ! -z "$key" ]; then
+            echo "            \"$key\": $value,"
+        fi
+    done | sed '$ s/,$//')
+
+    # Save stats to JSON
+    cat > "logs/${DATASET_ID}-${TODAY}_stats.json" << EOF
+{
+    "dataset_id": "$DATASET_ID",
+    "timestamp": "$(date -u +"%Y-%m-%dT%H:%M:%SZ")",
+    "stats": {
+        "markdown_files": $MARKDOWN_COUNT,
+        "html_files": $HTML_COUNT,
+        "ingest": {
+            "chunks_split": {
+                "pages": ${PAGES_COUNT:-0},
+                "chunks": ${CHUNKS_COUNT:-0}
+            },
+            "status": "$INGEST_STATUS"
+        },
+        "scrape": {
+$SCRAPE_PARSED
+        }
+    }
+}
+EOF
+
+    # Validate and pretty-print the JSON
+    if command -v jq >/dev/null 2>&1; then
+        jq '.' "logs/${DATASET_ID}-${TODAY}_stats.json" > "logs/${DATASET_ID}-${TODAY}_stats.json.tmp" && \
+        mv "logs/${DATASET_ID}-${TODAY}_stats.json.tmp" "logs/${DATASET_ID}-${TODAY}_stats.json"
```

Comment:
Why is the `mv` needed? Why not this:
```suggestion
        jq '.' "logs/${DATASET_ID}-${TODAY}_stats.json" > "logs/${DATASET_ID}-${TODAY}_stats.json"
```

Reply:
This characterizes the issue with doing jq directly:

https://stackoverflow.com/questions/68135435/output-redirection-produces-empty-file-only-with-jq

File becomes empty because shell truncates output before jq reads, temp file creation gets around this.

Reply:
Let me commit a suggestion